vm/vm_test.go

```diff
@@ -778,41 +779,67 @@ func runInterruptTest(t *testing.T, script string) {
 		t.Errorf("Define error: %v", err)
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
 	go func() {
 		<-waitChan
-		Interrupt(env)
+		cancel()
 	}()
 
-	_, err = env.Execute(script)
+	_, err = env.ExecuteContext(script, ctx)
 	if err == nil || err.Error() != ErrInterrupt.Error() {
 		t.Errorf("execute error - received %#v - expected: %#v", err, ErrInterrupt)
 	}
 }
 
-func TestInterruptConcurrency(t *testing.T) {
+func TestCancelChannelWithContext(t *testing.T) {
```

Comment:
Please look closely at where TestCancelChannelWithContext is getting interrupted at. I think using `time.Sleep(time.Millisecond)` is not going to get consistent results. If want to cancel at the correct time (at the channel wait) going to need to use a more complicated script. Look at TestCancelWithContext as to some hints on how.

Reply:
Updated